Assets/Scripts/BlasterPlayerScript.cs

```diff
@@ -32,24 +43,56 @@ void Update()
         {
             return;
         }
-        // Set animation
-        bool isMoving = input.move.x != 0 || input.move.y != 0;
-        animator.SetBool("isMoving", isMoving);
 
-        // Set scale for animation direction
-        if (isMoving)
+        // Get movement input
+        Vector2 moveInput = input.move;
+
+        // Initialize movement
+        Vector3 movement = Vector3.zero;
+
+        // Calculate movement based on input
+        if (moveInput.y > 0 && moveInput.x > 0) // W + D
```

Comment:
I always find this sorta code ugly but I don't think I can make it any better (cartesian code go brr